app/models/spree/gateway/adyen_credit_card.rb

```diff
@@ -19,5 +20,125 @@ def method_type
     def authorize(amount, card, gateway_options)
       ActiveMerchant::Billing::Response.new(true, "dummy authorization response")
     end
+
+    # Performs and authorization call to Adyen for the payment
+    # @raise [Spree::Core::GatewayError] if the encrypted card data is missing
+    # @raise [Spree::Core::GatewayError] if the authorize call fails
+    def authorize_new_payment payment
+      response = perform_authorization(payment)
+
+      unless response.success?
+        payment.log_entries.create!(details: response.to_yaml)
+        raise Spree::Core::GatewayError.new(
+          I18n.t(:credit_card_data_refused, scope: 'solidus-adyen')
+        )
+      end
+
+      payment.response_code = response.psp_reference
+      payment.save!
+      update_stored_card_data(payment)
+    end
+
+    private
+
+    def new_credit_card? source
+      source.encrypted_data.present?
+    end
+
+    def perform_authorization payment
+      # If this is a new credit card we should have the encrypted data
+      if new_credit_card?(payment.source)
+        rest_client.authorise_recurring_payment(
+          authorization_request(payment, true)
+        )
+      elsif payment.source.has_payment_profile?
+        rest_client.reauthorise_recurring_payment(
+          authorization_request(payment, false)
+        )
+      else
+        raise EncryptedDataError.new(
+          I18n.t(:missing_encrypted_data, scope: 'solidus-adyen')
```

Comment:
also I don't think it appropriate for errors to use i18n, errors are not intended to be user facing and there for should not use localizations.